### ISSUE TITLE
Rowspan processing is possibly buggy

### ISSUE THREAD

**Issue by Demosteneus** [speedata/publisher#259]
Hi,

I think the processing of rowspan (or in combination with colspan) might be buggy.
I have attached 3 data.xml:

**data0.xml** - prints correctly
Column 1 and 2 have rowspan over 3 rows. Then I have some colspans, which also works.
<a href="https://imgur.com/b8d0tDI"><img src="https://i.imgur.com/b8d0tDI.jpg" title="source: imgur.com" /></a>

Problems start as soon as I need to rowspan a later column. For example column 9, rows 2 and 3.
**data1.xml** - here cell from row 3, column 6 prints a column later, so there are some cells overlapping. More exactly, the spanned cell from column 9 overlaps, row 3 column 8.
<a href="https://imgur.com/DYB4bps"><img src="https://i.imgur.com/DYB4bps.jpg" title="source: imgur.com" /></a>

Finally, if I want to use rowspan on column 10 row 1 to row 3, this breaks completely, and no PDF is output, with error:
**data2.xml**
```
  <Error code='1'>[page 1] ...am Files\speedata-publisher\sw/lua/publisher\tabular.lua:1077: attempt to index local 'current' (a nil value)
stack traceback:
	...ram Files\speedata-publisher\sw/lua/publisher\spinit.lua:45: in function 'call'
	...ram Files\speedata-publisher\sw/lua/publisher\spinit.lua:376: in function 'main_loop'
	...ram Files\speedata-publisher\sw/lua/publisher\spinit.lua:387: in main chunk
	[C]: in function 'require'
	[\directlua]:1: in main chunk</Error>
```
<a href="https://imgur.com/woyX0ZV"><img src="https://i.imgur.com/woyX0ZV.jpg" title="source: imgur.com" /></a>

[Rowspan - buggy.zip](https://github.com/speedata/publisher/files/5014803/Rowspan.-.buggy.zip)

**Comment by pgundlach** [comment]
A slightly shorter example that triggers the error:

```xml
<data>
	<tr>
		<td rowspan="2" colspan="1"> 1/1, 1/2 </td>
		<td rowspan="1" colspan="4"> 3/1, 4/1, 5/1, 6/1 </td>
		<td rowspan="1" colspan="3"> 7/1, 8/1, 9/1</td>
		<td rowspan="2" colspan="1">10/1, 10/2</td>
	</tr>
	<tr>
		<td rowspan="1" colspan="2">3/2, 4/2</td>
		<td rowspan="1" colspan="2">5/2, 6/2</td>
		<td rowspan="1" colspan="2">7/2, 8/2</td>
		<td>9/2</td>
	</tr>
</data>
```

and

```xml
<Layout
  xmlns:sd="urn:speedata:2009/publisher/functions/en"
  xmlns="urn:speedata.de:2009/publisher/en" >
  <Trace grid="yes" />
  <SetGrid nx="10" height="12pt" />

  <Pageformat height="4cm" width="15cm" />

  <Record element="data">
    <PlaceObject column="1">
      <Table stretch="max" >
        <Columns>
          <Loop select="9">
            <Column width="1*"/>
          </Loop>
        </Columns>
        <ForAll select="tr">
          <Tr break-below="yes">
            <ForAll select="td">
              <Td padding="2pt" rowspan="{ @rowspan }" colspan="{ @colspan }" border-right="0.5pt" border-bottom="0.5pt">
                <Paragraph>
                  <Value select="."/>
                </Paragraph>
              </Td>
            </ForAll>
          </Tr>
        </ForAll>
      </Table>
    </PlaceObject>
  </Record>
</Layout>
```

**Comment by pgundlach** [comment]
Fixed in version 3.9.28, available for dl now. Thank you for the report!
